lib/UtapiClient.js

```diff
@@ -898,6 +899,70 @@ class UtapiClient {
             return undefined;
         });
     }
+
+    /**
+     *
+     * @param {string} location - name of data location
+     * @param {number} updateSize - size in bytes to update location metric by,
+     * could be negative, indicating deleted object
+     * @param {string} reqUid - Request Unique Identifier
+     * @param {function} callback - callback to call
+     * @return {undefined}
+     */
+    pushLocationMetric(location, updateSize, reqUid, callback) {
+        const log = this.log.newRequestLoggerFromSerializedUids(reqUid);
+        const params = {
+            level: 'location',
+            service: 's3',
+            location,
+        };
+        this._checkMetricTypes(params);
+        const action = (updateSize < 0) ? 'decrby' : 'incrby';
+        const size = (updateSize < 0) ? -updateSize : updateSize;
+        return this.ds[action](generateKey(params, 'locationStorage'), size,
+            err => {
+                if (err) {
+                    log.error('error pushing metric', {
+                        method: 'UtapiClient.pushLocationMetric',
+                        error: err,
+                    });
+                    return this._pushLocalCache(params, 'locationStorage', null,
+                        log, callback);
+                }
+                return callback();
+            });
+    }
+
+    /**
+     *
+     * @param {string} location - name of data backend to get metric for
+     * @param {string} reqUid - Request Unique Identifier
+     * @param {function} callback - callback to call
+     * @return {undefined}
+     */
+    getLocationMetric(location, reqUid, callback) {
```

Comment:
I would expect the metrics retrieval to be follow how Utapi lists other metrics (i.e., through a GET request to the API endpoint). This seems like a relatively big departure from the Utapi design.

Reply:
We made the decision to keep the location metric (both push & get) separate from the rest of Utapi functionality. It's definitely a departure from the original design, but it will be used differently from the original push & list. The get is going to be used by S3 regularly, so it makes more sense to create a method rather than continually making HTTP requests.